lib/bolt/config.rb

```diff
@@ -167,9 +188,20 @@ def update_from_cli(options)
       end
 
       TRANSPORT_OPTIONS.each do |key|
-        # TODO: We should eventually make these transport specific
         TRANSPORTS.each do |transport|
-          self[:transports][transport][key] = options[key] if options[key]
+          unless %i[ssl host_key_check].any? { |k| k == key }
+            self[:transports][transport][key] = options[key] if options[key]
+            next
+          end
+          if key == :ssl && transport == :winrm
```

Comment:
I don't fully understand what this section is doing, and why we weren't already doing something like this for options like `:key` or `:cacert`?

Reply:
We've generally been passing through options, even if they're unused by the transport. Eventually we'll want to refactor this a bit so transports can register their own expected options.